pkg/events/options.go

```diff
@@ -67,13 +67,17 @@ func WithVisibility(rights ...ttnpb.Right) Option {
 // WithAuthFromContext returns an option that extracts auth information from the context when the event is created.
 func WithAuthFromContext() Option {
 	return optionFunc(func(e *event) {
-		authentication := &ttnpb.Event_Authentication{}
 		if p, ok := peer.FromContext(e.ctx); ok && p.Addr != nil && p.Addr.String() != "pipe" {
 			if host, _, err := net.SplitHostPort(p.Addr.String()); err == nil {
-				authentication.RemoteIP = host
+				e.innerEvent.RemoteIP = host
 			}
 		}
```

Comment:
I think this now fits better in the "user agent" thing (and then maybe it's more ClientInfo than UserAgent)